Document where NServiceBus will look for license information, and in what order

DavidBoike opened the pull request:
Culled from [the code that is responsible](https://github.com/Particular/NServiceBus/blob/a151f2341d61a57ade9b9be29e7a9c08fbe4002f/src/NServiceBus.Core/Licensing/LicenseLocationConventions.cs) so should be accurate.

The one **DEPRECATED** I'm assuming because in the code it's referred to as `oldLocalLicenseFile`. Assume it's still being checked for backward compatibility.

DavidBoike commented:
Please review @Particular/engineers

SeanFeldman commented on nservicebus/licensing/license-management.md:
```diff
@@ -110,3 +110,22 @@ As a developer you can specify the license to use in your configuration code:
+
+If you are experiencing an issue with licensing, it can be helpful to know where NServiceBus is looking, and in what order. An expired license in `HKEY_CURRENT_USER`, for example, will overrule a valid license in `HKEY_LOCAL_MACHINE`. Note that these vary somewhat in older versions.
+
+In order to find the license file, NServiceBus will look:
```
> In order to find the license file,

How about just "license" since half of the locations are registry entry?

DavidBoike commented:
@SeanFeldman agree - changed.

SimonCropp commented:
@andreasohlund please review

andreasohlund commented:
Looks good

SimonCropp commented on nservicebus/licensing/license-management.md:
```diff
@@ -110,3 +110,22 @@ As a developer you can specify the license to use in your configuration code:
 
 <!-- import License -->
 
+## Where does NServiceBus look for a license?
```
so the above content references different version behaviour.

Is the below content correct for all versions of NSB?

SimonCropp commented on nservicebus/licensing/license-management.md:
```diff
@@ -110,3 +110,22 @@ As a developer you can specify the license to use in your configuration code:
+* License XML defined by `NServiceBus/License` appSetting
+* File path configured through `NServiceBus/LicensePath` appSetting
+* File located at `<BaseDirectory>\NServiceBus\License.xml`
+* **DEPRECATED** File located at `<BaseDirectory>\License\License.xml`
```
**DEPRECATED** seems redundant here. either it checks it or it doesnt

SimonCropp commented on nservicebus/licensing/license-management.md:
```diff
@@ -110,3 +110,22 @@ As a developer you can specify the license to use in your configuration code:
 
+## Where does NServiceBus look for a license?
+
+If you are experiencing an issue with licensing, it can be helpful to know where NServiceBus is looking, and in what order. An expired license in `HKEY_CURRENT_USER`, for example, will overrule a valid license in `HKEY_LOCAL_MACHINE`. Note that these vary somewhat in older versions.
```
better to go with 

> For example an expired license in `HKEY_CURRENT_USER` will overrule a valid license in `HKEY_LOCAL_MACHINE`. 

Putting for example in the middle causes someone to rethink the purpose of the statement after they have already made assumptions about the previous part.

SimonCropp replied:
> Note that these vary somewhat in older versions.

In what way?

SimonCropp replied:
> If you are experiencing an issue with licensing, it can be helpful to know where NServiceBus is looking, and in what order.

Kind of redundant  in the context of doco. just state what u r explaining. 

Get it "Kind of redundant"... I'll be here all week

SimonCropp commented on nservicebus/licensing/license-management.md:
```diff
@@ -110,3 +110,22 @@ As a developer you can specify the license to use in your configuration code:
+
+* License XML defined by `NServiceBus/License` appSetting
+* File path configured through `NServiceBus/LicensePath` appSetting
+* File located at `<BaseDirectory>\NServiceBus\License.xml`
```
so by `<BaseDirectory>` do u mean `Environment.CurrentDirectory` or the directory of the current running exe/dll? This often makes a difference in the context of windows services.

DavidBoike commented:
@SimonCropp Great notes! Think I'll probably lose context when I commit so will comment here.

* Intro paragraph: changed
* Versions - I put this at the end because I thought a nearly identical blob for each version would really muddy things. Decided to go spelunking through Git history and reorganize this all into a table.
* Deprecated: you're right, deprecation makes no sense, version-awareness is better.
* Replaced `<BaseDirectory>` with `{AppDomain.CurrentDomain.BaseDirectory}` which is what the code looks for. Do you think we need to call out what that means in different hosting contexts?
* "Kind of redundant" - have to admit, I chuckled

SimonCropp commented on nservicebus/licensing/license-management.md:
```diff
@@ -110,3 +110,28 @@ As a developer you can specify the license to use in your configuration code:
+
+In order to find the license, NServiceBus will examine:
+
+| Version | Location                                                                          | Notes |
```
table = awesome
